src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs

```diff
@@ -740,6 +740,9 @@ public void RefreshAndSearchAsync()
 
         internal void RefreshInfectedPackages()
         {
+            //do not call a protected route, if user is not logged in already.
+            if (string.IsNullOrEmpty(PackageManagerClientViewModel.Username)) return;
```

Comment:
wouldn't it be more idiomatic to check if the user is logged in? Or does that call actually start a login process as well?

Reply:
switched to `LoginState`